server.go

```diff
@@ -64,12 +64,17 @@ func handleComments(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 
-		// Write out the json response
-		commentsEncoder := json.NewEncoder(cFile)
-		commentsEncoder.Encode(comments)
+		// Pretty print the JSON, write to file
+		var encodedComments, _ = json.MarshalIndent(comments, "", "    ")
```

Comment:
1. I don't think you can actually do multiple assignment with var, only :=, but your throwing away the err so it maybe doesn't matter in this case? change it to be...

```go
encodedComments, err := json.MarshalIndent(comments, "", "    ")
if err != nil {
    http.Error(w, fmt.Sprintf("Unable to marshal comments to json: %s", err), http.StatusInternalServerError)
    return
}
```

2. See above

3. So yes, it would be GC'd. But, it's just generally bad practice to use the non streaming version. Since this is an example I'd rather be exemplifying the "right" (some would probably argue it's not), vs the most convenient. 

FWIW: The issue with using {Un,}Marhsal is that they read in the entire source and return byte buffers. So, pretend the json file is 50MB in size. So for each post request that would use the Marshal/Unmarshal pair (vs the encoders) the file would need to be read into a []byte, which would be ~50MB. Then you need to UnMarshal that into the []comment, which would be another data structure of ~50 MB modulo JSON, and then another 50MB+ []byte get's created when you MarshalIndent the updated comment back to JSON. So memory usage is ~150MB doing it this way. I haven't run any specific tests, but the pure Encoder/Decoder version of this will use ~50MB total, most of it because we realize the fully constructed []comment.

However this is an example, and ... AFAICT the other examples are doing it by just reading everything into ram / constructing arbitrary sized objects in ram and then the reverse to write data back out. So maybe I'm being too picky about the example?

I'd be happy to submit a pull request that switches everything to using UnMarshal/MarshalIndent instead if you think it's best.